core/src/main/java/org/elasticsearch/action/admin/cluster/allocation/ClusterAllocationExplanation.java

```diff
@@ -49,6 +54,9 @@
     private final Map<DiscoveryNode, Float> nodeWeights;
     private final UnassignedInfo unassignedInfo;
     private final long remainingDelayNanos;
+    private final List<IndicesShardStoresResponse.StoreStatus> shardStores;
+
+    private Map<DiscoveryNode, IndicesShardStoresResponse.StoreStatus> nodeStoreStatus = null;
```

Comment:
I wonder if we should use `@Transient` for this. I mean, we don't use java's serialization but it is kind of a useful signal.